components/constants/brave_services_key_v2.cc

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/constants/brave_services_key_v2.h"
+
+#include <vector>
+#include "base/logging.h"
+
+#include "base/base64.h"
+#include "base/strings/string_number_conversions.h"
+#include "base/strings/stringprintf.h"
+#include "brave/components/constants/buildflags.h"
+#include "crypto/hmac.h"
+#include "crypto/sha2.h"
+
+namespace constants {
+
+namespace {
+
+std::optional<std::string> GetServiceKey(Service service) {
+  if (service == Service::kAIChat) {
+    return BUILDFLAG(AI_CHAT_SERVICE_KEY);
+  }
+
+  return std::nullopt;
+}
+
+}  // namespace
+
+std::optional<std::pair<std::string, std::string>> GetBraveServicesV2Headers(
+    const std::string payload,
+    Service service) {
+  // Calculate the SHA-256 hash of the payload, then encode it in base64 format
+  // to generate the digest header. This also forms the signature string that
+  // needs to be signed.
+  std::string request_digest_base64;
```

Comment:
I think this should be broken out into it's own function and unit tested independently since there are test vector we can steal from https://www.ietf.org/archive/id/draft-ietf-httpbis-digest-headers-04.html#section-10.4 it seems we have code for this in a couple places, it'd be nice to consolidate them at some point ( though the following point may complicate that. )

also it appears that according to the spec it should be `sha-256=`, not `SHA-256=` - maybe we should switch our use.

Reply:
In this other doc https://datatracker.ietf.org/doc/html/draft-cavage-http-signatures#section-3.1 it's `SHA-256`, and that's what's used in the RequestSigner for rewards

Reply:
kk, i'm fine with leaving as-is